### PR TITLE
feat!: Add op impls, UTC helper function and interop with time crate for Duration

### PR DESCRIPTION
**Pull request by Techassi**
Part of https://github.com/stackabletech/operator-rs/issues/649

This PR adds op implementations between `Duration` and `std::time::Instant` and additionally adds the `now_utc` helper function.

It also adds interoperability with the `time` crate behind the new `time` feature. Enabling that feature will add basic math operations (`Add`, `AddAssign`, `Sub`, and `SubAssign`) between Duration and `time::OffsetDateTime`.

### CODE REVIEW

**Review by sbernauer — DISMISSED** [review]

File: Cargo.toml [hunk 1]
```diff
@@ -14,6 +14,9 @@ license.workspace = true
 edition.workspace = true
 repository.workspace = true
 
+[features]
+time = ["dep:time"]
```
Comment (sbernauer) [inline Cargo.toml:18]: What do you think of enabling this by default, as time is such a small crate (I guess)?

**Review by sbernauer — CHANGES_REQUESTED** [review]

File: src/time/mod.rs [hunk 2]
```diff
@@ -0,0 +1,7 @@
+mod duration;
```
[the remark this answers is not in the document]
Comment (Techassi) [reply src/time/mod.rs:1]: You are right, makes sense!

File: Cargo.toml (hunk 1 above)
Comment (Techassi) [reply Cargo.toml:18]: I made it opt-in because currently we need the interop only for one operator (secret-operator).

Comment (sbernauer) [reply Cargo.toml:18]: I know ;) Was just an idea to not need users to know this is behind a feature flag. But I don't care, was just an idea :)

File: src/time/mod.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (Techassi) [reply src/time/mod.rs:1]: Updated in ef93ca3

File: CHANGELOG.md [hunk 3]
```diff
@@ -21,11 +21,15 @@ All notable changes to this project will be documented in this file.
   - `ClusterCondition::is_good`
   - `ClusterOperationsConditionBuilder::new`
   - `commons::pdb::default_pdb_enabled`
-
-[#674]: https://github.com/stackabletech/operator-rs/pull/674
+- Add interoperability between the `time` crate and the `stackable_operator::time::Duration` struct. This is opt-in and
+  requires the `time` feature to be enabled. Additionally, adds `Add`, `AddAssign`, `Sub`, and `SubAssign` operations
+  between `Duration` and `std::time::INstant`. Further adds a new helper dunction `Duration::now_utc` which calculates
```
Comment (sbernauer) [inline CHANGELOG.md:26]:
```suggestion
  between `Duration` and `std::time::Instant`. Further adds a new helper function `Duration::now_utc` which calculates
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -21,11 +21,15 @@ All notable changes to this project will be documented in this file.
   - `ClusterCondition::is_good`
   - `ClusterOperationsConditionBuilder::new`
   - `commons::pdb::default_pdb_enabled`
-
-[#674]: https://github.com/stackabletech/operator-rs/pull/674
+- Add interoperability between the `time` crate and the `stackable_operator::time::Duration` struct. This is opt-in and
+  requires the `time` feature to be enabled. Additionally, adds `Add`, `AddAssign`, `Sub`, and `SubAssign` operations
+  between `Duration` and `std::time::Instant`. Further adds a new helper dunction `Duration::now_utc` which calculates
```
Comment (sbernauer) [inline CHANGELOG.md:26]:
```suggestion
  between `Duration` and `std::time::Instant`. Further adds a new helper function `Duration::now_utc` which calculates
```

**Review by sbernauer — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (Techassi) [reply CHANGELOG.md]: Waaay ahead of you, :) I already fixed it in dbca142

**Pull request merged**